Rename the `Import` class to `ImportSource`

agarny opened the pull request:
Addresses issue #189.

agarny committed:
Renamed the `Import` class to `ImportSource` (#189).

… and its private implementation from `ImportImpl` to
`ImportSourceImpl`.

agarny committed:
Renamed the `import.[cpp|h]` file to `importsource.[cpp|h]` (#189).

hsorby commented:
The OS X build slave appears to be having issues with it's connection to the internet.  This is under investigation and clearly not fixed, I thought this had been resolved but obviously I will need to dig further into why it is having problems.

You can try resending the request again, @nickerso knows how :).

hsorby commented on src/api/libcellml/component.h:
```diff
@@ -41,10 +41,10 @@ class LIBCELLML_EXPORT Component: public ComponentEntity
      * @param name The name of the component in the imported model to use.
      */
-    void setSourceComponent(const ImportPtr &imp, const std::string &name);
+    void setSourceComponent(const ImportSourcePtr &import, const std::string &name);
```
The parameter here should be named importSource.

agarny commented:
Thanks for the explanation @hsorby. I have no idea how to resend the request again, so will wait for @nickerso.

hsorby commented on src/api/libcellml/importsource.h:
```diff
@@ -22,23 +22,23 @@ limitations under the License.
  *
- * The import class is for encapsulating an external CellML Model.
+ * The import source class is for encapsulating an external CellML Model.
  */
```
We need to add more documentation here, @nickerso and I talked about this today I will try and add our thinking on this tomorrow.  I have forgotten and don't want to write the wrong thinking down.

nickerso replied:
I'm not one to let being wrong get in the way of writing a long comment on github :) hopefully the following is not so wrong that it points people in the wrong direction...@hsorby and I might revise tomorrow.

The basic idea is that the `ImportSource` does two things: the first is it points to a URL (the source) from which a CellML model is expected to be obtainable; the second is that it acts as a container/set/handle for all imported entities which should be obtained from the same *instance* of that source model. This becomes important, for example, when you have sibling components imported from the same instance (variable connections maintained) or different instances (connections severed). As with @hsorby's comment and from #188, I think a bit more of an explanation about what an `ImportSource` is would be useful here. I also now think "encapsulating an external CellML model" is actually incorrect/misleading.

Maybe @hsorby and I can come up with a better docstring tomorrow based on the above and comments on #188...

hsorby commented on src/api/libcellml/importsource.h:
```diff
@@ -22,23 +22,23 @@ limitations under the License.
      * @brief Set the source Model.
      *
-     * Set the source Model that this import refers to.
+     * Set the source Model that this import source refers to.
```
Again we should add more documentation to make this clearer about what the source model is and what we are doing.

hsorby reviewed (commented):
Hopefully @nickerso can clarify the points I have been vague on.

agarny committed:
Replaced the `imp`|`import` variable name with `importSource` (#189).

agarny commented:
Hmm... it looks like the macOS slave is now behaving...

hsorby commented:
Yes it is temperamental it usually works but sometimes the network access seems to fail for reasons unknown.

MichaelClerx commented:
Can we merge this or are we waiting for better docstrings?

nickerso commented:
Better docstrings would be good. I'll try to get time with @hsorby today to propose something.

nickerso committed:
Update imported source documentation.

Tweaking the documentation of the imported source class and its methods to better inform users as to its purpose.

nickerso committed:
Update import source to match new documentation.

Changing implementation's `mReference` to `mSource`.

agarny commented:
@nickerso's changes look good to me. So, @nickerso, if you are OK with my own changes, then feel free to merge this in (or someone else can merge this in if he's happy with all the changes).
